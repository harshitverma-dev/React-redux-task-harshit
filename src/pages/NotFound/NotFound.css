.not-found-page {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    min-height: 100vh;
    background: linear-gradient(135deg, #f5f7fa 0%, #e4e8eb 100%);
    padding: 20px;
}

.not-found-page h1 {
    font-size: 12rem;
    color: #dc3545;
    margin: 0;
    font-weight: 900;
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.1);
    animation: pulse 2s infinite;
}

.not-found-page h2 {
    font-size: 2.5rem;
    color: #343a40;
    margin: 1rem 0;
    font-weight: 600;
}

.not-found-page p {
    color: #6c757d;
    margin-bottom: 2rem;
    font-size: 1.2rem;
    max-width: 500px;
    text-align: center;
}

.home-page-link {
    padding: 15px 30px;
    background-color: #007bff;
    color: white;
    text-decoration: none;
    border-radius: 50px;
    font-weight: 600;
    text-transform: uppercase;
    letter-spacing: 1px;
    transition: all 0.3s ease;
    box-shadow: 0 4px 15px rgba(0, 123, 255, 0.2);
}

.home-page-link:hover {
    background-color: #0056b3;
    transform: translateY(-2px);
    box-shadow: 0 6px 20px rgba(0, 123, 255, 0.3);
}

@keyframes pulse {
    0% {
        transform: scale(1);
    }

    50% {
        transform: scale(1.05);
    }

    100% {
        transform: scale(1);
    }
}

@media (max-width: 768px) {
    .not-found-page h1 {
        font-size: 8rem;
    }

    .not-found-page h2 {
        font-size: 2rem;
    }

    .not-found-page p {
        font-size: 1rem;
        padding: 0 20px;
    }
}