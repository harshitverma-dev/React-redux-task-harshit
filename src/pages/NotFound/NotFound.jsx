import React from 'react';
import { Link } from 'react-router-dom';
import './NotFound.css';

const NotFound = () => {
  return (
    <div className="not-found-page">
      <h1>404</h1>
      <h2>Page Not Found</h2>
      <p>The page you are looking for doesn't exist or has been moved.</p>
      <Link to="/" className="home-page-link">
        Go to Product Listing
      </Link>
    </div>
  );
};

export default NotFound;