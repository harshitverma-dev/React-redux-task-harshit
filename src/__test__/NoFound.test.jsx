import React from 'react';
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NotFound from '../pages/NotFound/NotFound';

const renderWithRouter = (component) => {
    return render(
        <BrowserRouter>
            {component}
        </BrowserRouter>
    );
};

describe('NotFound Component', () => {
    it('renders 404 heading', () => {
        renderWithRouter(<NotFound />);
        expect(screen.getByText('404')).toBeInTheDocument();
    });

    it('renders error message', () => {
        renderWithRouter(<NotFound />);
        expect(screen.getByText('Page Not Found')).toBeInTheDocument();
        expect(
            screen.getByText("The page you are looking for doesn't exist or has been moved.")
        ).toBeInTheDocument();
    });

    it('renders home page link', () => {
        renderWithRouter(<NotFound />);
        const homeLink = screen.getByText('Go to Product Listing');
        expect(homeLink).toBeInTheDocument();
        expect(homeLink).toHaveAttribute('href', '/');
    });

    it('matches snapshot', () => {
        const { asFragment } = renderWithRouter(<NotFound />);
        expect(asFragment()).toMatchSnapshot();
    });
});