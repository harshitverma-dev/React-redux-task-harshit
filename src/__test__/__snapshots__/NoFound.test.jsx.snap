// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="not-found-page"
  >
    <h1>
      404
    </h1>
    <h2>
      Page Not Found
    </h2>
    <p>
      The page you are looking for doesn't exist or has been moved.
    </p>
    <a
      class="home-page-link"
      data-discover="true"
      href="/"
    >
      Go to Product Listing
    </a>
  </div>
</DocumentFragment>
`;
